### ISSUE TITLE
Remove boat shops from CTF?

### ISSUE THREAD

**Issue by AsuMagic** [transhumandesign/kag-base#1404]
**Current in-game behaviour.**

Boat shops exist in CTF.

**Proposed Change**

Remove boat shops from CTF.  
Map creators could still include pre-built (non-fixable) boats if they decide it would be interesting for early game rushes.

**Additional context**

Due to the gameplay dynamic in CTF, extremely few maps have any amount of water that would be useful for any kind of boat, except for maps with one or multiple large lakes.  
In TTH, this is generally feasible, because the halls mechanic makes larger maps bearable.  
In CTF, large maps tend to work extremely poorly in pubs because of goofy ways to cap flags, and large lakes force a single type of gameplay that generally feels very unpleasant and leads to stalemates.  
One of the very few compromises is to make multiple paths for combat (e.g. one lake on top and one big tunnel system on the bottom), which tends to split the action and makes gameplay more bearable.

Yet, even on the very few maps where there is enough water to place boats, those tend to go unused, and tend to not be a useful or interesting mechanic:
- Dinghies are almost useless on water. They have a few other uses:
    - To block flag bases. They're weird to interact with and more annoying than anything else. That (unintended) mechanic is not even interesting as a detail imo.
    - Glitch abuse through weird physics interactions.
    - Catapulting? Haven't seen anyone do this and it seemed pretty useless at best.
    - Dropping on people, who don't take damage (weird, but it would be OP if they did), can't pick them up, and get pushed down in a glitchy manner.
- Longboats suck **badly*:
    - There are better ways to cross average sized lakes for cheaper
    - Those are designed to be good at ramming towers, but ramming is usually not that helpful. You're just going to break some tiles any builder can fix and die immediately because being stuck under a tower does not put you at an advantage.
    - They're _extremely_ exposed, and there's nothing you can do about this. Just slash down people who are ramming and you've got a free multikill.
- Now, warboats are slightly more useful: They enable respawning, are somewhat more solid, and provide some amount of cover. But:
    - They cost gold (oof),
    - They're very easy to capture _anyway_, making them a really, _really_ dubious investment.
    - They're somewhat easy to break, though that could be buffed. Realistically, keg drops are going to ruin your day anyway.
    - Respawn interactions are glitchy as you can respawn in warboats being capped (in TTH at least, not sure about CTF), with spawn invincibility. Fixing this would make them even more trivial to capture.
    - They're practically useless for ramming.

Boats in general have seen extremely little unironic use in CTF matches.  
At this point, to me, boat shops are useless, and should not even be presented as an useful option for newbies at all.  
Keep the shops minimal, remove this category of clutter that has no gameplay purpose we can justify.

Arguments for keeping them, which I don't agree with:
- Most attempts at water CTF maps sucked, making good water CTF maps is possible.
- Bombarrows are too OP against boats, and limit their viability significantly.
- We should not remove such features entirely because of balance issues.
- Longboats are actually useful for ramming.

Should we keep water shops in CTF?

**Comment by mugg91** [comment]
I see Dinghy often used, even though not as intended but more for meme strats and wall clipping.
It would still be sad to remove it, since I remember having a lot of fun with it and I think it fits so well in the game.
It's in the game so many years now.
Also keep in mind boats are showcased in your trailer video. New players would be interested in boats.
Recently, I saw good use of longboat and warboat on the SharkRomance map.

So I would rather keep boats and look into ways to make boats not suck and not be glitchy.

Just throwing some ideas:

- Make it so you can't build blocks underwater or inside of water. 
Prevent building across bodies of water.
Encourages use of boats.

- Don't buff character movement in water [as done here](https://github.com/transhumandesign/kag-base/pull/1155). 
Imo, water movement wasn't so bad that characters needed a buff.
Keep characters slow in water while making boats fast.

- Similar to how doors eat sword attacks, make sword attacks from below boats not hit the driver.
Use raycast collision check, if boat is inbetween, don't hurt the driver.

- Slightly reduce the shape of Dinghy to make it not so glitchy in small areas.
Find a fix for "dinghy flying through the air wall ejection".
On detach, place the character at the base of the Dinghy (ground position) so they don't end up in ceilings.

Maybe make it so you can throw Dinghy to the ground only when there is space, make it do a "not allowed" sound in tight spaces (if walls are to the left and to the right).

- Make warboat not give attackers too many coins. This seems like a weak point.

- Increase time it takes to convert longboat and warboat.

- Increase health of longboat and warboat.

There is that change with [outpost](https://github.com/transhumandesign/kag-base/pull/1158).
You can't spawn at Ballista.
Not sure if you can still switch class at Ballista.
I'm saying it should be consistent with Warboat.
So if anything is removed from Ballista, remove it from Warboat, too.
Since Ballista doesn't cost gold, make Warboat not cost gold either.

**Comment by Gingerbeard5773** [comment]
Seems like this issue should be renamed to "Improve boats" rather than "Remove boats"!

**Comment by terracraft321** [comment]
> Seems like this issue should be renamed to "Improve boats" rather than "Remove boats"!

**Instead of removing boat shops from CTF, it would be better to focus on making more water-based maps and improving the boats' gameplay mechanics.**

- The current state of boats in CTF is unenjoyable.
- Instead of removing boat shops, focus on creating more water-based maps and improving boat gameplay mechanics.
- Aspire to make boats a valuable and enjoyable part of the game.
- Encourage boat warfare in CTF
- Improving boats and creating more water-based maps is a better solution than removing boat shops from CTF
- Make boats a valuable and enjoyable feature rather than a neglected and underwhelming one.

The current state of boats in CTF is one of unenjoyment and inutility. Rather than removing boat shops from the game, I propose that we focus on creating more water-based maps and improving the gameplay mechanics of boats. By doing so, we can transform boats from a hindrance to a source of enjoyment, and from an unnecessary burden to a valuable asset.

Let us not be content with simply addressing the issues with dinghies and the usefulness of longboats for ramming. Rather, let us aspire to elevate boats to their full potential, by creating more opportunities for their use and by refining their mechanics.

Though some water CTF maps may have been less than ideal in the past, that does not mean that we should abandon all hope of creating successful water-based maps. Instead, we should persevere in our efforts to bring forth the best water CTF maps that we're capable of. This means incentivizing boat warfare in CTF.

In conclusion, I believe that improving boats and creating more water-based maps is a better solution than simply removing boat shops from CTF. By doing so, we can make boats a valuable and enjoyable part of the game, rather than a neglected and underwhelming feature.

**Comment by ferrezinhre** [comment]
> > Seems like this issue should be renamed to "Improve boats" rather than "Remove boats"!
> 
> **Instead of removing boat shops from CTF, it would be better to focus on making more water-based maps and improving the boats' gameplay mechanics.**
> 
> * The current state of boats in CTF is unenjoyable.
> * Instead of removing boat shops, focus on creating more water-based maps and improving boat gameplay mechanics.
> * Aspire to make boats a valuable and enjoyable part of the game.
> * Encourage boat warfare in CTF
> * Improving boats and creating more water-based maps is a better solution than removing boat shops from CTF
> * Make boats a valuable and enjoyable feature rather than a neglected and underwhelming one.
> 
> The current state of boats in CTF is one of unenjoyment and inutility. Rather than removing boat shops from the game, I propose that we focus on creating more water-based maps and improving the gameplay mechanics of boats. By doing so, we can transform boats from a hindrance to a source of enjoyment, and from an unnecessary burden to a valuable asset.
> 
> Let us not be content with simply addressing the issues with dinghies and the usefulness of longboats for ramming. Rather, let us aspire to elevate boats to their full potential, by creating more opportunities for their use and by refining their mechanics.
> 
> Though some water CTF maps may have been less than ideal in the past, that does not mean that we should abandon all hope of creating successful water-based maps. Instead, we should persevere in our efforts to bring forth the best water CTF maps that we're capable of. This means incentivizing boat warfare in CTF.
> 
> In conclusion, I believe that improving boats and creating more water-based maps is a better solution than simply removing boat shops from CTF. By doing so, we can make boats a valuable and enjoyable part of the game, rather than a neglected and underwhelming feature.

About making more water based maps:

1. map length shouldn't be less than 300, because water body will take 1/3 of whole length, 400 okay go for it, 500 well iirc you all were screaming to remove "big maps".
2. any connected chains of islands above water bodies will render boats useless, any island on mid will require bigger lake, why? Well we have trampolines, catas, caged ratmens, + others things.
3. you will say that I can lower sky to prevent other ways to scale lake distance and force boats, yea you can I have nothing to add, genius idea, let force ppl to play how I want.

I really really want to see what water maps you all can make, so go make something!
